.github/workflows/ci.yml

```diff
@@ -18,13 +18,14 @@ jobs:
     steps:
     - uses: actions/checkout@v1
     - name: Cache Carthage dependencies
-      uses: actions/cache@v1.1.0
+      uses: actions/cache@v2
       id: carthage-cache
       with:
         path: Carthage
-        key: ${{ runner.os }}-carthage-stable-${{ hashFiles('**/Cartfile.resolved') }}
+        key: ${{ runner.os }}-carthage-cache-${{ hashFiles('**/Cartfile.resolved') }}
+        restore-keys: ${{ runner.os }}-carthage-cache
```

Comment:
We explicitly removed the restore key for carthage since it creates problems. When a dependency is removed, cache action restores the latest cache and runs carthage again. But carthage doesn't remove the cached dependency, and it plays along nicely. I think the issue was RxTesting framework.
So, we opted for totally re-building carthage cache if cartfile is updated.

Reply:
Something was wrong with the setup and it didn't work at all. Even though it found the cache with the given hash, the binaries inside weren't actually valid and Carthage rejected it. Then when it rebuild them, it didn't update the cache, because it already existed.

I remember the problem with the cache but for now, this is an OK tradeoff IMO. We can get all CI jobs finished in 2-5 minutes, instead of 15 minutes. It has also some $$$ consequences since we pay for every CI minute.

I'd suggest keeping it like this, instead of optimizing for the rare use case of removing a dependency.

Reply:
Hmmm, now when I think about it, the missing restore key + the issue with invalid framework versions are probably not related. I'll remove the restore keys.